### ISSUE TITLE
Add an ability to discuss a proposal privately

### ISSUE THREAD

**Issue by adulau** [MISP/MISP#822]
Add an ability to discuss a proposal privately. It's basically starting a discussion between a set of organizations.

**Comment by cvandeplas** [comment]
Why not doing this by other means than by using MISP? 
The contact reporter button was meant for this. 
Adding such a feature would add some more complexity layers above the discussions.

**Comment by adulau** [comment]
It's not the reporter. We want to contact the contributor of the proposal. How would you contact the respective proposal contributor for a specific proposal?

**Comment by iglocska** [comment]
The contact reporter in itself has a lot of shortcomings as it assumes that
who you are contacting is located on your instance. Otherwise the current
owner will be contacted.

**Comment by cvandeplas** [comment]
Ok, I understand it better. 
I think we need to stay consistent:
- either contact reporter or contributor out of band. (like it exists today, but extended and fixing limitations)
- either discuss using MISP

My preference would go to out of band as my fear would be to see MISP evolve into a complex (and thus higher security risk) discussion medium, instead of an technical information sharing platform.

**Comment by iglocska** [comment]
That makes sense. Right now MISP allows for some limited filtered
discussions, but only as long as it stays local on the instance (and is
therefore quite simplistic), indeed having cross instance discussions would
be a massive climb in complexity.

**Comment by adulau** [comment]
Discussing within MISP (privately and on local instance) is the most convenient way if you are a reverser or an analyst especially for the following reasons:

 -  When you have a lot of proposals on a single event, you don't know who to contact, especially with org having a lot of active users. It's usually better to be able to start a discussion on a proposal within the org members doing the proposal.

- Encryption is already present and convenient. Doing it out-of-band is bringing back the old issues of emailing and potential insecure emails. This was the goal of MISP to maintain a coherent flow of information and I think it's really the counter part of the "contact reporter".

- It's very common to ask a proposal reporter to create another event based on his/her proposals, so this is indeed a local instance feature. So the feature can be kept simple and only for the local users of an instance.

**Comment by cvandeplas** [comment]
@adulau : good arguments.

To keep in mind: We'll need to make sure the limitations caused by the lack of syncing discussions are clearly reflected to the user who wants to communicate.  (like not allowing communication for something that has no local org)

**Comment by adulau** [comment]
Good point. We should add something like "Discussions are not distributed and local to this MISP instance".

**Comment by iglocska** [comment]
Interesting ideas.

So as a solution for this would a new feature as described below work?

Have a mechanism that allows you to conveniently start a discussion with
the org of a proposal creator locally via the discussions system - we could
automagically create a non-selectable ad hoc sharing group for this that
only includes the two organisations. This would allow us to use the
existing mechanisms and it would also allow the creator of the discussion
to involve 3rd parties by editing the sharing group.
